### DISCUSSION TITLE
Issue when compiling WW3

### DISCUSSION THREAD

**Discussion by NS-Nik** [NOAA-EMC/WW3#695] [General]
Hi! 

I was following the Manual step by step and faced a problem on the compilation stage (Section 5.7)  

 When I am running the w3_automake or w3_make commands I get the same mistake: 

![Screenshot from 2022-04-28 19-23-29](https://user-images.githubusercontent.com/104583032/165800347-a446d5b5-c8a6-46ff-b543-92697e40bde3.png)
![Screenshot from 2022-04-28 19-23-42](https://user-images.githubusercontent.com/104583032/165800405-f83c92f2-87ca-4a97-a650-c2002fbce9fc.png)

**Comment by aliabdolali** [comment]
@NikitaS1990 
A couple of things:
Do you have required libraries like compiler, netcdf, mpi? loaded in your envirorment?
Did you do model/bin/setup ./model -c <comp>, so proper link and comp files are generated in you model/bin?
Do you have a switch in your model/bin directory?

You also can use cmake to compile to code, see [cmake Readme](https://github.com/NOAA-EMC/WW3/blob/develop/model/README.md)

**Reply by NS-Nik** [reply]
I have all necessary libraries and linked netcdf library. 

The environment was set up via w3_setup command: 
./model/bin/w3_setup model -c intel -s Ifremer1

comp, link and switch files are in the right directories.

**Comment by jaslanli** [comment]
Hi @NS-Nik, I'm facing with the same problem. Could you please share how did you fix it?

**Comment by gonesurfing** [comment]
I also ran into issues using the w3_automake script. The cmake method worked though. From the root directory, run this:
```
mkdir build  
cd build  
cmake ..  
make  

```
I did have a build fail with ubuntu 24.04. It works in 22.04. Something changed in gcc or gfortran between those releases.

**Reply by jaslanli** [reply]
Hi @gonesurfing, are you able to run regression tests after this installation? Could you please help me with [this](https://github.com/NOAA-EMC/WW3/discussions/1318)?

**Comment by mickaelaccensi** [comment]
@NS-Nik , you should use run_cmake_test instead of run_test for the regression tests.
For example, you can find some tutorials [here](https://gitlab.ifremer.fr/wave/Training/-/wikis/WW3-2023), like Tutorial WW3 compilation which include running a regression test.

Best

**Comment by NS-Nik** [comment]
@jaslanli Hey! 

It's been a long time since I asked this question :)

In this particular case the Intel compiler did not work in my environment. I tried Gnu compiler and it solved the problem. 

But now I use the new version of the WW and compile it with [cmake](https://github.com/NOAA-EMC/WW3/blob/develop/model/README.md)
